extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -458,19 +500,19 @@ export abstract class AzureAuth implements vscode.Disposable {
 		}
 	}
 
-	private async refreshAccessToken(account: azdata.AccountKey, rt: RefreshToken, tenant?: Tenant, resource?: Resource): Promise<TokenRefreshResponse> {
+	private async refreshAccessToken(account: azdata.AccountKey, rt: RefreshToken, tenant: Tenant = this.commonTenant, resource?: Resource): Promise<TokenRefreshResponse> {
 		const postData: { [key: string]: string } = {
 			grant_type: 'refresh_token',
 			refresh_token: rt.token,
 			client_id: this.clientId,
-			tenant: this.commonTenant,
+			tenant: tenant.id,
```

Comment:
I never realized I had made this mistake.... this might solve a bunch of azure connection issues.